### ISSUE TITLE
Adding tfa.models support - state-of-the-art models. 

### ISSUE THREAD

**Issue by innat** [tensorflow/addons#2409]
**Describe the feature and the current behavior/state.**

There are many new deep learning models that are not yet part of the `tf.keras.application`. So, it would be great if there is `tfa.models` support. 

That includes new state-of-the-art models with their last optimized weights such as `ResNeSt`, `GPU-Efficient-Networks`, `SENet`, `NFNet` etc. Additionally including more promising modeling ideas such as `CapsNet`, `ViT` etc. 

**Relevant information**
- Are you willing to contribute it (yes/no): yes, with other dedicated people 
- Are you willing to maintain it going forward? (yes/no): yes
- Is there a relevant academic paper? (if so, where): not sure 
- Is there already an implementation in another framework? (if so, where): [rwightman/pytorch-image-models](https://github.com/rwightman/pytorch-image-models) - timm

- Was it part of tf.contrib? (if so, where): not sure 

**Which API type would this fall under (layer, metric, optimizer, etc.)**: I guess, it's new. 

**Who will benefit from this feature?**: `tf.keras` users

**Any other info.** ---

**Comment by bhack** [comment]
We are evaluating how to better coordinate Addons with our model ecosystem. 
In the meant time I suggest you to start to contribute your model in https://github.com/tensorflow/models

/cc @tomerk
